### DISCUSSION TITLE
OpenAPI doc generation with Pydantic 2 examples is broken

### DISCUSSION THREAD

**Discussion by PabloReszczynski** [fastapi/fastapi#11137] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import fastapi
import pydantic
import yaml
from fastapi.openapi.utils import get_openapi

app = fastapi.FastAPI()


class Body(pydantic.BaseModel):
    name: str = pydantic.Field(
        ..., description="Name of the user", examples=["John", "Doe"]
    )
    age: int = pydantic.Field(..., description="Age of the user", examples=[18, 25])


@app.get("/")
def index(body: Body = fastapi.Body(...)):
    return body


print(
    yaml.dump(
        get_openapi(
            title=app.title,
            version=app.version,
            openapi_version=app.openapi_version,
            routes=app.routes,
        )
    )
)
```
```bash
$ python main.py
components:
  schemas:
    Body:
      properties:
        age:
          description: Age of the user
          examples:
          - 18
          - 25
          title: Age
          type: integer
        name:
          description: Name of the user
          examples:
          - John
          - Doe
          title: Name
          type: string
      required:
      - name
      - age
      title: Body
      type: object
    HTTPValidationError:
      properties:
        detail:
          items:
            $ref: '#/components/schemas/ValidationError'
          title: Detail
          type: array
      title: HTTPValidationError
      type: object
    ValidationError:
      properties:
        loc:
          items:
            anyOf:
            - type: string
            - type: integer
          title: Location
          type: array
        msg:
          title: Message
          type: string
        type:
          title: Error Type
          type: string
      required:
      - loc
      - msg
      - type
      title: ValidationError
      type: object
info:
  title: FastAPI
  version: 0.1.0
openapi: 3.1.0
paths:
  /:
    get:
      operationId: index__get
      requestBody:
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/Body'
        required: true
      responses:
        '200':
          content:
            application/json:
              schema: {}
          description: Successful Response
        '422':
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/HTTPValidationError'
          description: Validation Error
      summary: Index
```

### Description

With Pydantic 2 new breaking change for defining examples in fields, the OpenAPI generation breaks because the standard doesn't expect all models to have `examples` as its only available for endpoints [but not schemas or properties](https://swagger.io/docs/specification/adding-examples/).
But since Pydantic 2 now always generate `examples` instead of a single `example`, then the validation of the document fails.

<img width="744" alt="image" src="https://github.com/tiangolo/fastapi/assets/7925564/4f4fd2d7-d0bd-4713-a86b-eaa572827aad">

The expectation is that if a model or field have multiple examples, the OpenAPI spec should only contain one of them in the `example` property.

### Operating System

macOS

### Operating System Details

intel. macOS Sonoma 14.2.1

### FastAPI Version

0.109.2

### Pydantic Version

2.6.1

### Python Version

3.11.7

**Comment by markConklin** [comment]
running into this same issue. should we open an issue against the repo ?

**Comment by MrCreosote** [comment]
I suspect another symptom of this problem may be that the open api docs don't seem to display examples for pydantic fields:

```python
from fastapi import FastAPI
from pydantic import BaseModel, Field

app = FastAPI()


class Root(BaseModel):
    example: str = Field(example="thing")
    examples: str = Field(examples=["thing"])


@app.get("/", response_model=Root)
async def root() -> Root:
    return Root(example="foo", examples="bar")
```

![image](https://github.com/user-attachments/assets/950f3e66-f93f-4cf9-90f6-f909a8b336c2)

Based on [1] I would expect the examples to show up here.

This is on FastAPI 0.115.4, Pydantic 2.9.2

1. https://fastapi.tiangolo.com/tutorial/schema-extra-example/#field-additional-arguments

**Comment by MrCreosote** [comment]
In my hands `examples` in a `Field` now show up in the documentation (but aren't as readable as `example`, especially for examples that have lists or dicts) but `Path` and `Query` still require `example` for anything to show up in the docs

**Comment by YuriiMotov** [comment, marked as answer]
This has been fixed and works well in current version.

Regarding `examples` for `Query` \ `Path` \ `Header` \ `Cookie`, take a look at this answer: https://github.com/fastapi/fastapi/discussions/13447#discussioncomment-13238567
